source/material_model/visco_plastic.cc

```diff
@@ -373,8 +375,12 @@ namespace aspect
 
           // Step 5: limit the viscosity with specified minimum and maximum bounds
           composition_viscosities[j] = std::min(std::max(viscosity_yield, min_visc), max_visc);
+
+          // Fill output parameters
+          output_parameters.composition_yielding = composition_yielding;
```

Comment:
I think these lines should be outside of this loop, but they will not be needed if you fill the vectors in the output_parameters structure directly.